include/cuco/detail/static_map/static_map.inl

```diff
@@ -237,13 +237,21 @@ template <class Key,
           class ProbingScheme,
           class Allocator,
           class Storage>
-template <typename OutputIt>
-OutputIt
+template <typename KeyOut, typename ValueOut>
+std::pair<KeyOut, ValueOut>
 static_map<Key, T, Extent, Scope, KeyEqual, ProbingScheme, Allocator, Storage>::retrieve_all(
-  OutputIt output_begin, cuda_stream_ref stream) const
+  KeyOut keys_out, ValueOut values_out, cuda_stream_ref stream) const
```

Comment:
Why do we need to use separate key and value iterators instead of using a key-value pair iterator?

Reply:
This would not introduce the construction of `cuco::pair` to users and relax the requirement of the iterator `value_type` must be constructible from the container's `value_type`.